modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/dump/WriteOnlyZipFileIO.java

```diff
@@ -17,142 +17,57 @@
 
 package org.apache.ignite.internal.processors.cache.persistence.snapshot.dump;
 
-import java.io.BufferedOutputStream;
-import java.io.File;
 import java.io.IOException;
+import java.io.OutputStream;
 import java.nio.ByteBuffer;
-import java.nio.MappedByteBuffer;
-import java.nio.channels.Channels;
-import java.nio.channels.WritableByteChannel;
-import java.nio.file.Files;
-import java.nio.file.Paths;
 import java.util.zip.ZipEntry;
 import java.util.zip.ZipOutputStream;
-import org.apache.ignite.internal.processors.cache.persistence.file.AbstractFileIO;
 import org.apache.ignite.internal.processors.cache.persistence.file.FileIO;
-import org.apache.ignite.internal.util.typedef.internal.A;
+import org.apache.ignite.internal.processors.cache.persistence.file.FileIODecorator;
 
 import static java.util.zip.Deflater.BEST_COMPRESSION;
-import static org.apache.ignite.internal.processors.cache.persistence.file.FilePageStoreManager.ZIP_SUFFIX;
 
 /**
  * {@link FileIO} that allows to write ZIP compressed file.
  * It doesn't support reading or random access.
  * It is not designed for writing concurrently from several threads.
  */
-public class WriteOnlyZipFileIO extends AbstractFileIO {
+public class WriteOnlyZipFileIO extends FileIODecorator {
```

Comment:
Let's inherit it from BufferedFileIO instead. It makes this class usage much simpler.